File: java/series/src/main/java/Series.java
import java.util.ArrayList;
import java.util.List;

class Series {
    private String string = "";

    Series(String string) {
        this.string = string;
    }

    List<String> slices(int sliceLength) {
        checkForValidSliceLength(sliceLength);

        List<String> foundSeries = new ArrayList<String>();
        for (int i = 0; i <= string.length() - sliceLength; i++) {
            foundSeries.add(string.substring(i, i + sliceLength));
        }

        return foundSeries;
    }

    private void checkForValidSliceLength(int sliceLength) {
        if (sliceLength > string.length()) {
            throw new IllegalArgumentException("Slice size is too big.");
        }

        if (sliceLength <= 0) {
            throw new IllegalArgumentException("Slice size is too small.");
        }
    }
}

File: solutions/csharp/series/2/Series.cs
public static class Series
{
    public static string[] Slices(string numbers, int sliceLength)
    {
        if (sliceLength <= 0 || sliceLength > numbers.Length)
        {
            throw new ArgumentException();
        }

        var slices = new List<string>();

        for (var i = 0; i < numbers.Length - sliceLength + 1; i++)
        {
            slices.Add(new string(numbers.Substring(i, sliceLength)));
        }

        return [.. slices];
    }
}